gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/KafkaEndpointConnector.java

```diff
@@ -182,9 +183,8 @@ protected Completable subscribe(final ExecutionContext ctx) {
 
                                     String groupId = getGroupId(ctx);
                                     config.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
-                                    String instanceId = UUID.randomUUID().toString();
-                                    config.put(ConsumerConfig.GROUP_INSTANCE_ID_CONFIG, instanceId);
-                                    config.put(ConsumerConfig.CLIENT_ID_CONFIG, instanceId);
+                                    config.put(ConsumerConfig.GROUP_INSTANCE_ID_CONFIG, KAFKA_CLIENT_IDENTIFIER);
```

Comment:
> A unique identifier of the consumer instance provided by the end user. 
> Only non-empty strings are permitted. If set, the consumer is treated as a static member, which means that only one instance with this ID is allowed in the consumer group at any time.
> This can be used in combination with a larger session timeout to avoid group rebalances caused by transient unavailability (e.g. process restarts). If not set, the consumer will join the group as a dynamic member, which is the traditional behavior.

Reading that documentation make me wonder that if I start SSE curl for the same application, it will change the current behavior where the 2 curl will consume a subset of the messages to only one of the curl command will consume all the messages. Is it something we really want (maybe optionally?)

Reply:
I thought about that, and I suggest we removed this settings because in our contexte the endpoint instance is shared accross external client and so is not static.

Reply:
OK, maybe group_intance_id is stil something to consider later if someone requires it.
Also, I can see that the kafka client id is always the same here. QoS is not something applied to producer but it is for consumer so what happen if we create 2 different consumers with the same client id  (because the QoS config changes) ?